servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/HttpInstrumentationHelper.java

```diff
@@ -0,0 +1,186 @@
+
+        serverInstrumenterBuilder
+                .addAttributesExtractor(HttpServerAttributesExtractor
+                        .builder(ServiceTalkHttpAttributesGetter.SERVER_INSTANCE)
+                        .setCapturedRequestHeaders(options.capturedRequestHeaders())
+                        .setCapturedResponseHeaders(options.capturedResponseHeaders())
+                        .build());
+        if (options.enableMetrics()) {
+            serverInstrumenterBuilder.addOperationMetrics(HttpServerMetrics.get());
+        }
+
+        Instrumenter<RequestInfo, io.servicetalk.http.api.HttpResponseMetaData> instrumenter =
+                serverInstrumenterBuilder.buildServerInstrumenter(RequestHeadersPropagatorGetter.INSTANCE);
+
+        return new HttpInstrumentationHelper(instrumenter, false);
+    }
+
+    /**
+     * Creates an HTTP client instrumentation helper.
+     *
+     * @param openTelemetry the OpenTelemetry instance
+     * @param options OpenTelemetry configuration options
+     * @param componentName component name for peer service attribute
+     * @return client instrumentation helper
+     */
+    static HttpInstrumentationHelper createClient(OpenTelemetry openTelemetry, OpenTelemetryOptions options,
+                                                  String componentName) {
+        SpanNameExtractor<RequestInfo> clientSpanNameExtractor =
+                HttpSpanNameExtractor.create(ServiceTalkHttpAttributesGetter.CLIENT_INSTANCE);
+        InstrumenterBuilder<RequestInfo, io.servicetalk.http.api.HttpResponseMetaData> clientInstrumenterBuilder =
+                Instrumenter.builder(openTelemetry, INSTRUMENTATION_SCOPE_NAME, clientSpanNameExtractor);
+        clientInstrumenterBuilder
+                .setSpanStatusExtractor(ServicetalkSpanStatusExtractor.CLIENT_INSTANCE)
+                .addAttributesExtractor(new DeferredHttpClientAttributesExtractor(options));
+
+        if (options.enableMetrics()) {
+            clientInstrumenterBuilder.addOperationMetrics(HttpClientMetrics.get());
+        }
+        componentName = componentName.trim();
+        if (!componentName.isEmpty()) {
+            clientInstrumenterBuilder.addAttributesExtractor(
+                    AttributesExtractor.constant(PEER_SERVICE, componentName));
+        }
+
+        Instrumenter<RequestInfo, io.servicetalk.http.api.HttpResponseMetaData> instrumenter =
+                clientInstrumenterBuilder.buildClientInstrumenter(RequestHeadersPropagatorSetter.INSTANCE);
+
+        return new HttpInstrumentationHelper(instrumenter, true);
+    }
+
+    private static final class DeferredHttpClientAttributesExtractor implements
+            AttributesExtractor<RequestInfo, io.servicetalk.http.api.HttpResponseMetaData> {
+
+        private final AttributesExtractor<RequestInfo, io.servicetalk.http.api.HttpResponseMetaData> delegate;
+
+        private DeferredHttpClientAttributesExtractor(OpenTelemetryOptions openTelemetryOptions) {
```

Comment:
```suggestion
        DeferredHttpClientAttributesExtractor(OpenTelemetryOptions openTelemetryOptions) {
```
bcz it's used outside of itself